src/widgets/clear-all/clear-all.js

```diff
@@ -70,17 +72,20 @@ function clearAll({
 
   return {
     init({helper, templatesConfig}) {
-      this._clearRefinementsAndSearch = clearRefinementsAndSearch.bind(null, helper);
+      this.helper = helper;
       this._templateProps = prepareTemplateProps({defaultTemplates, templatesConfig, templates});
     },
 
     render({results, state, createURL}) {
-      let hasRefinements = getRefinements(results, state).length !== 0;
+      this.clearAttributes = getRefinements(results, state)
+        .map(one => one.attributeName)
+        .filter(one => excludeAttributes.indexOf(one) === -1);
+      let hasRefinements = this.clearAttributes.length !== 0;
       let url = createURL(clearRefinementsFromState(state));
 
       ReactDOM.render(
         <ClearAll
-          clearAll={this._clearRefinementsAndSearch}
+          clearAll={this.clearAll.bind(this)}
```

Comment:
`bind` should not be called in render. It's best to store the bound function on `this`, and do that in the `init` method.